libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -2146,3 +2151,31 @@ void NavEKF3_core::verifyTiltErrorVariance()
     }
 }
 #endif
+
+/*
+  move the EKF origin to the current position at 1Hz. The public_origin doesn't move.
+  By moving the EKF origin we keep the distortion due to sperical
+  shape of the earth to a minimum.
+ */
+void NavEKF3_core::moveEKFOrigin(void)
+{
+    // only move origin when we have a origin and we're using GPS
+    if (!frontend->common_origin_valid || !filterStatus.flags.using_gps) {
+        return;
+    }
+
+    // move the origin to the current state location
+    Location loc = EKF_origin;
+    loc.offset(stateStruct.position.x, stateStruct.position.y);
+    const Vector2F diffNE = loc.get_distance_NE_ftype(EKF_origin);
+    EKF_origin = loc;
+
+    // now fix all output states
+    stateStruct.position.xy() += diffNE;
```

Comment:
I assume the reason this is calculated like this rather than being set to zero is becasue of rounding errors at lines 2169 and 2170

Reply:
yes, the lat,lon is to approx 1cm resolution, so we need to ensure we don't add 1cm of position noise